screens/EventDetails/TimeSection.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react"
+import { StyleSheet, TouchableOpacity, View } from "react-native"
+import * as Calendar from 'expo-calendar';
+import { Headline, Caption, CaptionTitle } from "@components/Text"
+import { dayjs, now } from "@lib/date"
+import { CalendarEvent, addToCalendar, getCalendar } from "@lib/Calendar"
+import { ButtonStyles, showToast } from "@lib/ButtonStyle"
+import { Ionicon } from "@components/common/Icons"
+
+interface TimeSectionProps {
+  color: string
+  event: CalendarEvent
+}
+
+const TimeSection = ({ color, event }: TimeSectionProps) => {
+  const [status, requestPermission] = Calendar.useCalendarPermissions();
+  const [addedEvent, setAddedEvent] = useState(false)
+  const startDateFormat = event.duration.formattedDate(
+    now(),
+    dayjs(event.duration.startDate)
+  )
+  const endDateFormat = event.duration.formattedDate(now(), dayjs(event.duration.endDate))
+  const startTimeFormat = dayjs(event.duration.startDate).format("h A")
+  const endTimeFormat = dayjs(event.duration.endDate).format("h A")
+  const hitSlopInset = {
+    top: 10,
+    bottom: 10,
+    right: 10,
+    left: 10
+  }
+
+  const addEventToCalendar = async () => {
+    let calendar: Calendar.Calendar
+
+    if (status?.granted) {
+      calendar = await getCalendar()
+      await addToCalendar(calendar.id, event, setAddedEvent)
+
+    } else {
+      await requestPermission().then(async (status) => {
+        if (status?.granted) {
+          calendar = await getCalendar()
+          await addToCalendar(calendar.id, event, setAddedEvent)
+        }
+      }).catch((error) => {
+       showToast("Unable to add Event without Permissions", event.bottomTabHeight)
+      })
+    }
+  }
+
+  return (
+    <View style={[styles.flexRow, styles.paddingIconSection]}>
```

Comment:
I think we could have some basic ui layout component that could be shared across these sections. Then if the layout changes in the future, it'll be easy to apply the styling changes to all sections. Don't worry about rn though.